packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -196,18 +203,25 @@ class OnsetPicker extends React.Component {
    * @param {ageUnit} - New ageUnit
    */
   changeAgeUnit(ageUnit) {
-    this.setState((prevState) => {
+    if (this.state.age !== undefined) {
+      this.setState((prevState) => {
       // Check if date can be calculated
-      const ageDate = Number.isInteger(prevState.age) && ageUnit
-        ? moment(this.props.birthdate).add(prevState.age, ageUnit) : undefined;
-      // Check if date is valid
-      const validDate = ageDate && ageDate >= moment(this.props.birthdate) && ageDate <= moment();
+        const ageDate = Number.isInteger(prevState.age) && ageUnit
+          ? moment(this.props.birthdate).add(prevState.age, ageUnit) : undefined;
+        // Check if date is valid
+        const validDate = ageDate && ageDate >= moment(this.props.birthdate) && ageDate <= moment();
```

Comment:
so instead of if, else block in this method, can we do this ?

 const validDate = (this.state.age !== undefined) ? (ageDate && ageDate >= moment(this.props.birthdate) && ageDate <= moment()) : undefined

Reply:
Since this.state.age is the same as prevState.age  leaving this line as it is.

Reply:
in that case can we use like this by declaring one more variable.

const age = (prevState.age !== undefined) ? Number.isInteger(prevState.age) : undefined
const ageDate = age && ageUnit ? moment(this.props.birthdate).add(prevState.age, ageUnit) : undefined;
const validDate = (this.state.age !== undefined) ? (ageDate && ageDate >= moment(this.props.birthdate) && ageDate <= moment()) : undefined

Reply:
Yes this works fine . Changed in 8461b22